snorkel/analysis/metrics.py

```diff
@@ -53,9 +53,6 @@ def metric_score(
         msg = f"The metric you provided ({metric}) is not currently implemented."
         raise ValueError(msg)
 
-    if filter_dict is None:
```

Comment:
note: without this change, `filter_dict` was _always_ getting set, and the following if statement was not used as intended https://github.com/HazyResearch/snorkel/pull/1347/files#diff-7556b7b31429c528d90b68a0b2a3473dR62

Reply:
great catch. should we add a test? or was external behavior valid?

Reply:
behavior was valid—it was just confusing in the code